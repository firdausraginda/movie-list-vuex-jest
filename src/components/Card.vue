<template>
  <div class="card col-4">
    <div class="card-body">
      <img
        class="card-img-top"
        :src="dataCard.imgFilm"
        alt="Card image cap"
      >
      <h5 class="card-title">{{cutTextJudul(dataCard.judulFilm)}}</h5>
      <p>{{cutTextDeskripsi(dataCard.deskripsiFilm)}}</p>
      <a
        class="btn btn-primary"
        @click="moveToDetail()"
      >Detail</a>
      <a
        v-if="dataCard.choosen == false"
        class="btn"
        @click="chooseDataFil()"
      >Pilih</a>
      <a
        v-if="this.$route.name == 'personal' && dataCard.choosen == true"
        class="btn"
        @click="deleteDataFil()"
      >Hapus</a>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      idFilmx: ""
    };
  },
  props: {
    dataCard: {
      type: Object,
      default: null
    }
  },
  methods: {
    cutTextJudul(text) {
      return `${text.substr(0, 30)}`;
    },
    cutTextDeskripsi(text) {
      return `${text.substr(0, 100)}..`;
    },
    moveToDetail() {
      this.$router.push({ path: `/detailFilm/${this.dataCard.idFilm}` });
    },
    chooseDataFil() {
      this.$store.dispatch("addDataFilmChoosen", this.dataCard);
    },
    deleteDataFil() {
      this.$store.dispatch("deleteDataFilmChoosen", this.dataCard);
    }
  }
};
</script>

<style lang="scss" scoped>
.btn-primary {
  color: white !important;
}
.card {
  border: none;
  text-align: left;
  img {
    border-radius: 5px;
    height: 200px;
    object-fit: cover;
    margin-bottom: 20px;
  }

  .card-title {
    font-weight: bold;
  }
}
</style>