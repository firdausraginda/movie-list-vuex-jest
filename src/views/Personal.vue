<template>
  <div class="personal">
    <div class="container">
      <div class="row">
        <Card
          v-for="film in getDaftarFilmChoose"
          :key="film.idFilm"
          :dataCard="film"
        />
      </div>
    </div>
  </div>
</template>

<script>
import Card from "../components/Card.vue";

export default {
  data() {
    return {};
  },
  components: {
    Card
  },
  computed: {
    getDaftarFilmChoose() {
      return this.$store.state.daftarFilmChoose;
    }
  }
};
</script>

