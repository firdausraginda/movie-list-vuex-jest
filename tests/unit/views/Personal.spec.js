import { shallowMount } from '@vue/test-utils'
import Personal from '../../../src/views/Personal.vue'
import Card from '../../../src/components/Card.vue'

describe('Personal.spec.js', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Personal, {
            computed: {
                getDaftarFilmChoose() {
                    return [{ idFilm: 'gans', choosen: false, imgFilm: 'https://upload.wikimedia.org/wikipedia/en/thumb/5/53/Black_mirror_bandersnatch_poster.jpg/220px-Black_mirror_bandersnatch_poster.jpg', judulFilm: 'Black Mirror: Bandersnatch', deskripsiFilm: 'Black Mirror: Bandersnatch adalah film interaktif tahun 2018 dalam seri antologi fiksi ilmiah Black Mirror. Film ini ditulis oleh pencipta serinya, Charlie Brooker, dan disutradarai oleh David Slade. Netflix merilisnya sebagai film terpisah pada tanggal 28 Desember 2018.' }]
                }
            }
        })
    })

    test('is Personal component',  () => {
        expect(wrapper.is(Personal)).toBe(true)
        expect(wrapper.is('.personal')).toBe(true)
    })

    test('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    test('contains a Card component', () => {
        expect(wrapper.contains(Card)).toBe(true)
    })

    test('Card element exists', () => {
        expect(wrapper.find(Card).exists()).toBe(true)
    })

    test('Personal has a "dataCard" property', () => {
        expect(wrapper.find(Card).vm.dataCard.idFilm).toEqual('gans')
    })

})