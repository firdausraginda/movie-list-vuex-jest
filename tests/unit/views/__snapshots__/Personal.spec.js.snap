// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal.spec.js has the expected html structure 1`] = `
<div
  class="personal"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <card-stub
        datacard="[object Object]"
      />
    </div>
  </div>
</div>
`;
